transport.go

```diff
@@ -74,21 +75,36 @@ func (c *GRPCClient) GetEmbedAccessToken(ctx context.Context, req *embedaccessto
 }
 
 func (c *GRPCClient) getConn(ctx context.Context) (*grpc.ClientConn, error) {
+	// TODO: when the interface of the speakeasy middleware instantiation is changed to enable an error to be propagated
+	//       to the callee, create the connection inline with the middleware instantiation.
+	c.Lock()
+	defer c.Unlock()
+	if c.conn == nil {
+		conn, err := createConn(context.Background(), c.secure, c.serverURL, c.grpcDialer)
```

Comment:
as you are creating a new context here, the one passed in as a parameter to this method can be removed?